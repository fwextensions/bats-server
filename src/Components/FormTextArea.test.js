import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import FormTextArea from './FormTextArea';

test('Snapshot', () => {
  const component = renderer.create(
    <FormTextArea
      label="Other"
      onChange={() => {}}
      property="otherObservationNotes"
      value='val'
    />);

    const componentJson = component.toJSON();
    expect(componentJson).toMatchSnapshot();
});

test('FormTextArea should trigger the onChange callback', () => {
  const mock = jest.fn();

  const wrapper = render(
    <FormTextArea
      label="Other"
      onChange={mock}
      property="otherObservationNotes"
      value='val'
    />);

    const input = wrapper.getByLabelText('Other');
    fireEvent.change(input, { target: { value: "t" }});
    fireEvent.change(input, { target: { value: "e" }});
    fireEvent.change(input, { target: { value: "s" }});
    fireEvent.change(input, { target: { value: "t" }});
    expect(mock.mock.calls.length).toBe(4);
})