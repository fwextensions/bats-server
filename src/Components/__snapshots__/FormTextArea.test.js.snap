// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Array [
  <label
    className="usa-label"
    htmlFor="otherObservationNotes"
  >
    Other
  </label>,
  <textarea
    className="usa-textarea"
    id="otherObservationNotes"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={false}
    value="val"
  />,
]
`;
